imports/lib/schemas/chat_notifications.ts

```diff
@@ -0,0 +1,51 @@
+import * as t from 'io-ts';
+import { date } from 'io-ts-types';
+import SimpleSchema from 'simpl-schema';
+import { BaseCodec, BaseOverrides } from './base';
+import { Overrides, buildSchema, inheritSchema } from './typedSchemas';
+
+// A notification triggered by a chat message sent by a user.
+const ChatNotificationFields = t.type({
+  // The userId of the user for whom this notification targets.
+  user: t.string,
+
+  // The sender of the chat message that tripped the user's dingwords.
+  // System messages are forbidden from triggering dingword notifications.
+  sender: t.string,
+
+  // The puzzle to which this chat was sent.
+  puzzle: t.string,
+  // The hunt in which the puzzle resides.
+  hunt: t.string,
+  // The message body. Plain text.
+  text: t.string,
```

Comment:
I had kind of assumed we'd record a foreign key to the chat message but I guess this makes it easier to track/deal with

Reply:
Yeah, it made slightly more sense to me to denorm it?  I guess you could always make the reactive subscribe do even more chaining, but given that the actions you can take are limited to "view the puzzle", it's not *that* useful to have a reference to a specific message.